### ISSUE TITLE
Shared constraints are not applied to database schema

### ISSUE THREAD

**Issue by droggo** [grails/grails-core#9649]
Shared constraints work fine for runtime validation but are not applied to table definitions for `run-app` and `schema-export`. For example having domain:

```
class Error {

	String code
	BigDecimal number

    static constraints = {
    	code shared: 'errorCode'
    	number shared: 'decimalPrecision'
    }
}
```
and shared constraints in `application.groovy`:
```
grails.gorm.default.constraints = {
	decimalPrecision(scale:7, max: new BigDecimal(10))
	errorCode(maxSize: 3, unique: true)
}
```
Will use following `ddl` to create database and `ddl.sql` file:
```

    drop table error if exists;

    create table error (
        id bigint generated by default as identity,
        version bigint not null,
        code varchar(255) not null,
        number decimal(19,2) not null,
        primary key (id)
    );
```
Error are:
- missing unique index
- not valid size of code column
- not valid scale of numeric column

If we define the same constraints on domain directly
```
static constraints = {
    code maxSize: 3, unique: true
    number scale:7, max: new BigDecimal(10)
}
```
correct DDL is generated:
```
    drop table error if exists;

    create table error (
        id bigint generated by default as identity,
        version bigint not null,
        code varchar(3) not null,
        number decimal(19,7) not null,
        primary key (id)
    );

    alter table error 
        add constraint UK_5j64usn3qht222h55mygbqr5v unique (code);
```

Still both versions of domain will be validated fine by GORM validation

**Comment by davidsonnabend** [comment]
Still having the problem using grails 3.1.3 on Mac OS X.
Set the following defaul constraint:

```
grails.gorm.default.constraints = {
    '*'(nullable: true)
}
```
Having the following Domain:

```
class Demo {
	String property01
	String property02

    static constraints = {
    }
}
```

I tried to validate and saved an demo instance:

```
def demo = new Demo(property01: "P01")
assert demo.validate()
assert demo.save()
```
Validation works but saving the instance results in an ` org.h2.jdbc.JdbcSQLException`: 
```
NULL not allowed for column "PROPERTY02"
```

**Comment by jeffbrown** [comment]
@graemerocher Was this fix included in 3.1.2 and/or 3.1.3?

**Comment by graemerocher** [comment]
shared constraints were included in 3.1.2 yes
